scripts/tools/zap/version_update.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This automates ZAP version changes as the zap version is repeated
+# in many places
+
+import logging
+import os
+import re
+import sys
+
+import click
+import coloredlogs
+
+# Supported log levels, mapping string values required for argument
+# parsing into logging constants
+__LOG_LEVELS__ = {
+    'debug': logging.DEBUG,
+    'info': logging.INFO,
+    'warn': logging.WARN,
+    'fatal': logging.FATAL,
+}
+
+# A version is of the form: v2023.01.09-nightly
+#
+# At this time we hard-code nightly however we may need to figure out a more
+# generic version string once we stop using nightly builds
+ZAP_VERSION_RE = re.compile(r'v(\d\d\d\d)\.(\d\d)\.(\d\d)-nightly')
+
+# A list of files where ZAP is maintained. You can get a similar list using:
+#
+# rg v2023.01.09-nightly --hidden --no-ignore --files-with-matches
+#
+# Excluding THIS file and excluding anything in .environment (logs)
+#
+# Set as a separate list to not pay the price of a full grep as the list of
+# files is not likely to change often
+FILES_DEPENDING_ON_ZAP_VERSION = [
+    '.github/workflows/build.yaml',
+    '.github/workflows/darwin-tests.yaml',
+    '.github/workflows/darwin.yaml',
+    '.github/workflows/fuzzing-build.yaml',
+    '.github/workflows/tests.yaml',
+    'integrations/docker/images/chip-build/Dockerfile',
+    'integrations/docker/images/chip-cert-bins/Dockerfile',
```

Comment:
Unfortunately, the actual update process here is:

1.  Update the Dockerfiles.
2. Oncer dockerhub is updated, as a separate PR update all the YAML files (to use the new Docker versions and the new ZAP version).

Reply:
Updated the script to have control onwhat to update: dockerfile or usage (or both if needed for some reason).